### DISCUSSION TITLE
Introduce friction when hitting backspace

### DISCUSSION THREAD

**Discussion by murphpdx** [JedWatson/react-select#5670] [Ideas]
We would like to make it harder to remove an item from a multi-select to make sure that the user intended to delete the option. If the option is not focused, the user will have to click backspace twice.

**Comment by murphpdx** [comment]
https://github.com/JedWatson/react-select/pull/5669

**Comment by Rall3n** [comment]
Hello @murphpdx,

something like this is possible using the `onKeydown` prop and a reference object.

```tsx
const ref = React.useRef<SelectInstance | null>(null);

<Select
  ref={ref}
  isMulti
  onKeyDown={(e) => {
   if (!['Delete', 'Backspace'].includes(e.key)) return;
   if (ref.current!.props.inputValue || ref.current!.state.focusedValue) return;
    
   ref.current!.focusValue('previous');
   e.preventDefault();
  }}
/>;
```

**Reply by murphpdx** [reply]
It doesn't seem to register the key-down event for me when it's deleting an item. I'm using `<CreatableSelect` if that matters?

**Reply by murphpdx** [reply]
```    
      case 'Delete':
      case 'Backspace':
        console.log(event.key);
        if (chipsRef.current?.props.inputValue || chipsRef.current?.state.focusedValue) {
          return;
        }
        chipsRef.current!.focusValue('previous');
        event.preventDefault();
        return;
      default:
        return;
```

https://github.com/JedWatson/react-select/assets/7244569/b83bd81c-4cd3-4849-bcfd-9f1bd3d93042

**Reply by murphpdx** [reply]
nvm I figured it out. It still would be a nice feature in the main component.
